x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -112,5 +126,33 @@ MlDailyMaintenanceService getDailyMaintenanceService() {
         return mlDailyMaintenanceService;
     }
 
+    private void makeMlIndicesHidden() {
+        String[] mlHiddenIndexPatterns = MachineLearning.getMlHiddenIndexPatterns();
+        GetSettingsResponse getSettingsResponse =
+            client.admin().indices().prepareGetSettings()
+                .setIndices(mlHiddenIndexPatterns)
+                .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN_CLOSED_HIDDEN)
+                .get();
+        String[] nonHiddenIndices =
+            getSettingsResponse.getIndexToSettings().stream()
+                .filter(e -> e.getValue().getAsBoolean(SETTING_INDEX_HIDDEN, false) == false)
+                .map(Map.Entry::getKey)
+                .toArray(String[]::new);
+        if (nonHiddenIndices.length == 0) {
+            logger.info("There are no indices that need to be made hidden, " + getSettingsResponse);
```

Comment:
After you've finished testing I think this should be downgraded to `DEBUG` level. It won't be of interest to many people.

```suggestion
            logger.debug("There are no ML indices that need to be made hidden, " + getSettingsResponse);
```